native/app/components/ListItem.js

```diff
@@ -2,53 +2,116 @@ import React from 'react';
 import {
   View,
   Text,
-  TouchableOpacity
+  TouchableOpacity,
+  StyleSheet
 } from 'react-native'
 
+import {
+  Title,
+  Subheading,
+  TouchableRipple } from 'react-native-paper';
+
+import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
+import { COLOR_NOTES_BLUE } from '../utils/constants';
+import moment from 'moment';
+
 const striptags = require('striptags');
 
-function formatLastModified(date) {
+function formatLastModified(date = new Date()) {
 
-  if (new Date().getDate() === date.getDate()) {
-    return date.toLocaleTimeString([], {
-      hour: '2-digit',
-      minute: '2-digit'
-    });
-  }
+  var m = moment(date);
 
-  return date.toLocaleDateString([], {
-    year: 'numeric',
-    month: 'long',
-    day: 'numeric'
-  });
+  if (m.isSame(new Date(), 'day')) {
+    return m.format('LT');
+  } else if (m.isSame(new Date(), 'year')) {
+    return m.format('ll');
+  } else {
+    return m.format('ll');
+  }
 }
 
 export default class ListItem extends React.Component {
+
+  constructor (props) {
+    super(props);
+
+    this._navigateToNote = () => {
+      props.navigate('EditorPanel', { note: props.note });
+    }
+  }
+
+
   render() {
     const {
-      content,
-      navigate,
-      rowId,
-      lastModified
+      note
     } = this.props;
 
+    let firstLine = '', secondLine = '';
+    // FIXME: not perfect, need to be properly done but is good for testing.
+    if (note.content) {
+      firstLine = striptags(note.content.replace('&nbsp;', ' ').split('</')[0]).substr(0, 150);
+      secondLine = striptags(note.content.replace('&nbsp;', ' ').replace(firstLine, '')).substr(0, 150);
+    }
 
     return (
-      <TouchableOpacity onPress={() =>
-        // TODO: Fix navigation here to go the the selected note using note uuid?
-        navigate('EditorPanel', {rowId: rowId})
-      }>
-        <View>
-          <View style={{ padding: 10 }} >
-            <Text>
-              {striptags(content).substr(0, 50)}
-            </Text>
-            <Text>
-              {formatLastModified(lastModified)}
-            </Text>
+      <TouchableRipple onPress={this._navigateToNote} >
```

Comment:
I don't see the Ripple actually happen? Do we need extra styles or use TouchableOpacity ?

Reply:
I need to investigate further why ripple do not happen when onPress have a navigate event. If it's not easy to fix we'll move back to TouchableOpacity.

Reply:
Going back with TouchableOpacity 👍